### ISSUE TITLE
Update `import_` utility?

### ISSUE THREAD

**Issue by mattwthompson** [mdtraj/mdtraj#1882]
This module has some magic for helpfully soft-importing optional modules and/or warning users when something isn't found. It hasn't been updated in a while and could use some refreshing. A few of the modules aren't so heavy (or likely to be installed by something else anyway) and might be easier to just package as required dependencies

https://github.com/mdtraj/mdtraj/blob/1.9.9/mdtraj/utils/delay_import.py

**Comment by chrisjonesBSU** [comment]
I figured I could a look at this. It looks like `scripttest` and `netCDF4` aren't actually used anywhere? There's a [note](https://github.com/mdtraj/mdtraj/blob/43d8194bbc2e7f97a74b69fa02708bf17589529e/docs/whatsnew.rst?plain=1#L556) from 2014 about removing netCDF4 and using scipy's implementation of netCDF instead. 

IMO, I think `pandas` `networkx` and `pytables` all fit the criteria of either being light enough, or likely already installed by other packages, so they can be included as required dependencies. That leaves `openmm` as probably the only optional dependency. 

I can start a PR that cleans up `delay_import.py`.

**Comment by chrisjonesBSU** [comment]
I just now saw https://github.com/mdtraj/mdtraj/pull/1878 which adds `netCDF4` back. It seems like this could be installed by default as well.

**Comment by mattwthompson** [comment]
Yeah - now that I actually look at the keys of the dictionary, this is a small enough module that paring it down and out of existence might be easiest.

* I don't know what `scripttest` is and it doesn't come up in the codebase, so let's remove references to it
* `networkx` is pretty light, so if it's used in more than a couple places we can just require it
* Similar story for `scipy`, I think
* Pandas is actually a fairly heavy one, and it is set to get much heavier in 3.0 with the Arrow bindings, so I'd prefer it stay optional
* I forget what `pytables` does, maybe it could be refactored out (in a separate PR)?
* OpenMM should stay optional IMO

In the day job we have a [little decorator](https://github.com/openforcefield/openff-utilities/blob/main/openff/utilities/utilities.py#L69-L85) that can wrap functions which need an optional dependency

```python
@requires_package("pandas")
def do_dataframe_thing(): ...
```

This loses context and installation instructions - but do we really need to include all of that here?

**Comment by mattwthompson** [comment]
This bit is IMO a little funky and using the built-in `warnings` module would be more consistent with standard practice: https://github.com/mdtraj/mdtraj/blob/43d8194bbc2e7f97a74b69fa02708bf17589529e/mdtraj/utils/delay_import.py#L188-L191

**Comment by jeremyleung521** [comment]
> Yeah - now that I actually look at the keys of the dictionary, this is a small enough module that paring it down and out of existence might be easiest.
> 
>     * I don't know what `scripttest` is and it doesn't come up in the codebase, so let's remove references to it
> 
>     * `networkx` is pretty light, so if it's used in more than a couple places we can just require it
> 
>     * Similar story for `scipy`, I think
> 
>     * Pandas is actually a fairly heavy one, and it is set to get much heavier in 3.0 with the Arrow bindings, so I'd prefer it stay optional
> 
>     * I forget what `pytables` does, maybe it could be refactored out (in a separate PR)?
> 
>     * OpenMM should stay optional IMO
> 
> 
> In the day job we have a [little decorator](https://github.com/openforcefield/openff-utilities/blob/main/openff/utilities/utilities.py#L69-L85) that can wrap functions which need an optional dependency
> 
> ```python
> @requires_package("pandas")
> def do_dataframe_thing(): ...
> ```
> 
> This loses context and installation instructions - but do we really need to include all of that here?

Pytables is the literal backbone for the HDF5 portions of the code. ([mdtraj/formats/hdf5.py](https://github.com/mdtraj/mdtraj/blob/934b2732aa7aaab3299c086b2f5abbe7013184ec/mdtraj/formats/hdf5.py#L283), [mdtraj/formats/lh5.py](https://github.com/mdtraj/mdtraj/blob/934b2732aa7aaab3299c086b2f5abbe7013184ec/mdtraj/formats/lh5.py#L411), [mdtraj/io.py](https://github.com/mdtraj/mdtraj/blob/934b2732aa7aaab3299c086b2f5abbe7013184ec/mdtraj/io.py#L86))

**Comment by janash** [comment]
@Chinwendu20 - I do not think this is fully addressed. I updated some information about `netcdf` with my PR #1878. However some of the messages in the `MESSAGES` dictionary appear to have not been updated for several years. For example, I don't know that we'd recommend `easy_install` anymore - more likely `pip` or `pandas`.

I do think it seems to be a discussion here of having a larger structural change (the decorator @mattwthompson mentioned) though I'm not aware of it (if it is something that happened, perhaps someone who knows can comment!). I'd be fine with an update of `MESSAGES` at the moment.

**Comment by mattwthompson** [comment]
What I was originally trying to hint at was indeed a larger structural change, one which hasn't happened or AFAICT been thought through.

We really should ditch mention of `easy_install`, it's ... a touch out of date 😅
